src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -287,26 +415,117 @@ int goto_analyzer_parse_optionst::doit()
   if(set_properties())
     return 7;
 
-  if(cmdline.isset("show-intervals"))
+
+  // Output file factory
+  std::ostream *out;
+  const std::string outfile=options.get_option("outfile");
+  if(outfile=="-")
+    out=&std::cout;
+  else
   {
-    show_intervals(goto_model, std::cout);
-    return 0;
-  }
+    out=new std::ofstream(outfile);
```

Comment:
Should be opened in binary mode (``std::ios::binary``) when later writing a goto binary to it.